lib/rest-api.php

```diff
@@ -11,14 +11,24 @@
 }
 
 /**
- * Registers the Global Styles REST API routes.
+ * Overrides the REST controller for the `wp_global_styles` post type.
+ *
+ * @param array  $args      Array of arguments for registering a post type.
+ *                          See the register_post_type() function for accepted arguments.
+ * @param string $post_type Post type key.
+ *
+ * @return array Array of arguments for registering a post type.
  */
-function gutenberg_register_global_styles_endpoints() {
-	$global_styles_controller = new WP_REST_Global_Styles_Controller_Gutenberg();
-	$global_styles_controller->register_routes();
-}
-add_action( 'rest_api_init', 'gutenberg_register_global_styles_endpoints' );
+function gutenberg_override_global_styles_endpoint( array $args ): array {
+	$args['rest_controller_class']           = 'WP_REST_Global_Styles_Controller_Gutenberg';
+	$args['revisions_rest_controller_class'] = 'Gutenberg_REST_Global_Styles_Revisions_Controller_6_6';
+	$args['late_route_registration']         = true;
+	$args['show_in_rest']                    = true;
+	$args['rest_base']                       = 'global-styles';
```

Comment:
The CI tests are running 6.5.5. I think it's computed for CI [here](https://github.com/WordPress/gutenberg/blob/43d70f9af460406bab79c6a79bd4323a81e93766/.github/workflows/unit-test.yml#L95-L95).

They should probably be running 6.6, but even so, adding the args for backwards compat just in case.

Compare 6.5 https://github.com/WordPress/wordpress-develop/blob/6.5/src/wp-includes/post.php#L473

With 6.6 https://github.com/WordPress/wordpress-develop/blob/6.6/src/wp-includes/post.php#L476

Reply:
> They should probably be running 6.6

I'm assuming that these tests run on the previous major WP version due to our BC commitment. That said, I did expect we also ran the unit tests on the latest too.